internal/resources/cluster/tkgservicevsphere/resource_tkg_service_vsphere.go

```diff
@@ -424,11 +516,11 @@ var info = &schema.Schema{
 	MaxItems:    1,
 	Elem: &schema.Resource{
 		Schema: map[string]*schema.Schema{
-			clusterNameKey: {
+			nodepoolNameKey: {
 				Type:        schema.TypeString,
 				Description: "Name of the nodepool",
 				Default:     defaultNodePoolName,
-				Optional:    true,
+				Required:    true,
```

Comment:
Default and Required cannot be set . 

`// If Required is true above, then Default cannot be set. DefaultFunc
	// can be set with Required. If the DefaultFunc returns nil, then there
	// will be no default and the user will be asked to fill it in.`

Reply:
Please revert this change and regenerate the doc

Reply:
Thanks for pointing this out @shreyassreenivas, we were missing on this as we were running the acceptance test for AWS resource and checking only the spec of that resource.